### ISSUE TITLE
The multi-controlled gate by RZGate().control() is less optimized.

### ISSUE THREAD

**Issue by tamiyaonodera** [Qiskit/qiskit#13473]
### Environment

- **Qiskit version**: 1.3.0rc2
- **Python version**: 3.12.6
- **Operating system**: Sonoma 14.5

### What is happening?

The multi-controlled gate by RZGate().control() is less optimized.

### How can we reproduce the issue?

We create a circuit with a 6-qubit multi-controlled RZGate.  We then transpile it to an Eagle device, and count the number of entangling gates.

```
from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager
backend = service.backend("ibm_sherbrooke")
pm = generate_preset_pass_manager(optimization_level=3, backend=backend, seed_transpiler=12345)

import numpy as np
from qiskit import QuantumCircuit, QuantumRegister
from qiskit.circuit.library import RZGate
n=6;  ang=np.pi/7
qr = QuantumRegister(n)
qc = QuantumCircuit(qr)
qc.append(RZGate(ang).control(n-1), qr)

isa_qc=pm.run(qc)
print(isa_qc.count_ops()['ecr'])
```

### What should happen?

The above code outputs 266 in 1.3.0rc2, while 65 in 1.2.4.

**Comment by tamiyaonodera** [comment]
I confirmed that this has not been fixed with 1.3.0.  It outputs 266.

**Comment by mtreinish** [comment]
Yes, the fix merged on main and backported in preparation for a 1.3.1 release which will probably be published next week. You can try installing qiskit from source on the stable branch with: `pip install 'git+https://github.com/Qiskit/qiskit@stable/1.3'`. You will need the rust compiler installed to do this.
